core/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -182,20 +188,14 @@ protected WritePrimaryResult shardOperationOnPrimary(BulkShardRequest request, I
                 // if its a conflict failure, and we already executed the request on a primary (and we execute it
                 // again, due to primary relocation and only processing up to N bulk items when the shard gets closed)
                 // then just use the response we got from the successful execution
-                if (item.getPrimaryResponse() == null || isConflictException(failure) == false) {
-                    item.setPrimaryResponse(new BulkItemResponse(item.id(), docWriteRequest.opType(),
+                if (replicaRequest.getPrimaryResponse() == null || isConflictException(failure) == false) {
```

Comment:
I will open a followup issue, for some reason github is erroring out on creating issues now :0